### ISSUE TITLE
[Telemetry] Enabling search suggestions

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#6070]
#### Description & Product Manager / Data Scientist User Story
- As a product owner, I want to know if users prefer to keep search suggestions disabled in private mode

#### What questions will you answer with this data?
- Do users choose to receive search suggestions in private mode (and therefore share their searches with their search provider)? How often do users choose this setting will determine what should be our default. 

#### What probes (suggested, if applicable)
- ~Enable "search in private mode" via settings menu~  (Already added in #975)
- Enable "search in private mode" via UI prompt (waiting for #1019)

### Dependencies
- [ ] Search suggestions opt-in #1019 

### Acceptance Criteria (Added by PM)
- Event pings can be queried via re:dash
- We are sending telemetry events for the actions listed in the requirements
- We have documented the telemetry
- We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry
- NOT an AC: Data science to create dashboard or further graphs (this will be a separate issue, this issue is only about hooking up the events described and that they can be queried in re-dash)

### Comment 0 changes
- 11-11 @Baron-Severin updated AC

**Comment by Baron-Severin** [comment]
This is waiting on #1019, which will add the UI element that requires telemetry.

**Comment by mcarare** [comment]
@vesta0 Just to clarify: we only want to track events of enabling search suggestions in private from the hint panel (UI prompt)? That is a unique event. 
Or we also want to track enabling from the settings screen? This could be a repeating event for the same user. TY!

**Comment by mcarare** [comment]
@liuche can you please help me with this info? Thank you!

**Comment by liuche** [comment]
> @vesta0 Just to clarify: we only want to track events of enabling search suggestions in private from the hint panel (UI prompt)? That is a unique event.

Yes, what users choose for this hint.

> Or we also want to track enabling from the settings screen? This could be a repeating event for the same user. TY!

Yes, we'd like to track this toggle event, and this setting should also be included in the baseline ping. See https://github.com/mozilla-mobile/fenix/issues/975#issuecomment-482523587

**Comment by AndiAJ** [comment]
Hi, verified as fixed on both **Fenix Nightly Build #13391806 from 12/5** and **Beta 3.0.0-beta.3 Build #13392157 from 12/5** using a Google Pixel 3a (Android 9)

► Beta
✔️ Choose "Allow" from hint - Ping **3f02c223-f241-4542-9d05-cbd3d071d9e5**
✔️ State when opening Search Settings  (updated and displayed properly) - Ping **fcdab8a9-4b25-4264-990d-c54d693908e2**
Toggle events:
✔️OFF - Ping **8be36d6c-3edf-4593-8936-74475b476f74**
✔️ON - Ping **be2b9cba-5dad-4354-b51e-acd3757fb80e**

[Logcat](https://github.com/mozilla-mobile/fenix/files/3931680/PBSU.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-test061200)

► Nightly
✔️ Choose "Allow" from hint - Ping **22095256-fbec-4bcd-8768-4e3f86f1dd02**
✔️ State when opening Search Settings  (updated and displayed properly) - Ping **969c94c1-ff7e-494d-9322-ba3e5812fbd3**
Toggle events:
✔️ OFF - Ping **9f6f49ee-ebe0-442d-a1c1-0fa0a5d97f58**
✔️ ON - Ping **d91bf209-637b-4028-a498-93d87299007c**
[Logcat](https://github.com/mozilla-mobile/fenix/files/3931681/PBSUNI.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-test0612000)

@liuche - Shouldn't we also track the "Don't allow" event from the hint ?
Or we track it only when disabling it from the Search Settings?
